Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -641,12 +645,18 @@ WarpXParticleContainer::DepositCharge (WarpXParIter& pti, RealVector const& wp,
         amrex::LayoutData<amrex::Real>* costs = WarpX::getCosts(lev);
         amrex::Real* cost = costs ? &((*costs)[pti.index()]) : nullptr;
 
-        ablastr::particles::deposit_charge<WarpXParticleContainer>
-            (pti, wp, ion_lev, rho, icomp, nc, offset, np_to_depose,
-             local_rho[thread_num], lev, depos_lev, this->charge,
-             WarpX::nox, WarpX::noy, WarpX::noz, ng_rho, dx, xyzmin, ref_ratio,
-             cost, WarpX::n_rz_azimuthal_modes, WarpX::load_balance_costs_update_algo,
-             WarpX::do_device_synchronize);
+        AMREX_ALWAYS_ASSERT(WarpX::nox == WarpX::noy);
+        AMREX_ALWAYS_ASSERT(WarpX::nox == WarpX::noz);
```

Comment:
I'm wondering if we can do this just once at initialization, rather than every time we deposit the charge. I know it's not expensive, but if we can check this once for all initially, I would be in favor of not replicating the check here. What do you think?

Reply:
This is done once per time step and costs nothing.
I do this here because at this point I throw away `nox` and `noy` when calling the next function.

We can generally refactor out the three variables to be one, since we don't support this anymore.

Reply:
OK, sounds good then.